src/index.js

```diff
@@ -37,51 +36,47 @@ io.on('connection', async (socket) => {
     })
     if (error) throw error
 
-    const context = {
-        user,
-        room,
-    }
-
-    socket.join(context.room)
+    const context = { user, room }
+    socket.join(room)
     socket.emit('message', generateWelcome())
-    socket.to(context.room).emit('message', generateJoined(user))
+    socket.to(room).emit('message', generateJoined(user))
 
-    const roomWithUsers = await roomUsersFetcher.call(room)
-    io.to(room).emit('roomData', roomWithUsers)
+    const syncRoomData = async () => {
+        const roomWithUsers = await roomUsersFetcher.call(room)
+        io.to(room).emit('roomData', roomWithUsers)
+    }
 
     console.log(`User ${username} has been joined to ${roomTitle}`)
+    syncRoomData()
 
-    socket.on('message', async (message, callback) => {
-        const newMessage = await messageToRoomSender.call({
-            user: context.user,
-            room: context.room,
-            message,
+    socket.on('message', async (text, callback) => {
+        const message = await messageToRoomSender.call({
+            ...context,
+            message: text,
             type: 'text',
         })
         io.to(context.room).emit('message', {
```

Comment:
How does this work?
I mean that `on('message')` a `emit('message')` happens? O_o

Reply:
i didn't get you

Reply:
Inside a 'reaction' to an event you emit the same event you're 'reacting' to.